### PR TITLE
Bomberman Boots

### PR DESCRIPTION
**Pull request by TylerJaacks**
This isn't fully complete it is a lot of hacks to bypass function but it allows for Bomberman to boot. There are some functions in here that are complete. Also I would like some help with the CF* classes, as well if I could upstream some of my changes and improve my hacks. It should boot to the loading screen the reason that it doesn't move past that I assume has to do with CFRunLoop* related code not being full complete.

### CODE REVIEW

File: src/frameworks/foundation/ns_file_manager.rs [hunk 1]
```diff
@@ -142,6 +145,45 @@ pub const CLASSES: ClassExports = objc_classes! {
     }
 }
 
+- (bool) changeCurrentDirectoryPath:(id)path {
```
Comment (TylerJaacks) [inline src/frameworks/foundation/ns_file_manager.rs:148]: - (bool) changeCurrentDirectoryPath:(id)path should be complete.

File: src/frameworks/foundation/ns_file_manager.rs [hunk 2]
```diff
@@ -142,6 +145,45 @@ pub const CLASSES: ClassExports = objc_classes! {
...
+            log!("Warning: chdir({:?}) failed, could not change working directory to {:?}, returning -1", path_str, path);
+
+            return false;
+        }
+    }
+
+    return true;
+}
+
+- (id) currentDirectoryPath {
```
Comment (TylerJaacks) [inline src/frameworks/foundation/ns_file_manager.rs:171]: - (id) currentDirectoryPath should be complete.

File: src/frameworks/foundation/ns_data.rs [hunk 3]
```diff
@@ -146,6 +146,19 @@ pub const CLASSES: ClassExports = objc_classes! {
     );
 }
 
+-(()) getBytes:(MutPtr<u8>)buffer {
```
Comment (TylerJaacks) [inline src/frameworks/foundation/ns_data.rs:149]: -(()) getBytes:(MutPtr<u8>)buffer this function should be complete.

**Comment by ciciplusplus** [comment]
Hello @TylerJaacks,

Please follow this guide https://github.com/hikari-no-yume/touchHLE/blob/trunk/CONTRIBUTING.md if you want to contribute.

In particular:
- you should push your commits to Gerrithub with Change-Ids
- please abide to https://github.com/hikari-no-yume/touchHLE/blob/trunk/dev-docs/code-style.md
- please separate your PR to smaller commits. For example, you should separate complete implementation from incomplete one
- please remove profanities from commit messages 👼 

Thanks

**Pull request closed without merging**